pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator_test.go

```diff
@@ -0,0 +1,199 @@
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	networkv1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1"
+	fake "k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned/fake"
+)
+
+const (
+	Group                = "networking.gke.io"
+	GKENetworkParamsKind = "GKENetworkParams"
+	// Default Network
+	DefaultGKENetworkParamsName = "DefaultGKENetworkParams"
+	DefaultVPCName              = "projects/testProject/global/networks/default"
+	DefaultVPCSubnetName        = "projects/testProject/regions/us-central1/subnetworks/default"
+	DefaultSecondaryRangeA      = "RangeA"
+	DefaultSecondaryRangeB      = "RangeB"
+	// Red Network
+	RedNetworkName          = "Red-Network"
+	RedGKENetworkParamsName = "RedGKENetworkParams"
+	RedVPCName              = "projects/testProject/global/networks/red"
+	RedVPCSubnetName        = "projects/testProject/regions/us-central1/subnetworks/red"
+	RedSecondaryRangeA      = "RedRangeA"
+	RedSecondaryRangeB      = "RedRangeB"
+)
+
+func TestPerformMultiNetworkCIDRAllocation(t *testing.T) {
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{Name: "node0"},
+		Spec:       v1.NodeSpec{ProviderID: ""},
+	}
+	testCases := []struct {
+		desc                       string
+		networks                   []*networkv1.Network
+		gkeNwParams                []*networkv1.GKENetworkParams
+		interfaces                 []*compute.NetworkInterface
+		wantDefaultNwPodCIDRs      []string
+		wantNorthInterfaces        networkv1.NorthInterfacesAnnotation
+		wantAdditionalNodeNetworks networkv1.MultiNetworkAnnotation
+	}{
+		{
+			desc: "default network only - should return default network cidrs and no multi-network annotations",
+			networks: []*networkv1.Network{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: networkv1.DefaultNetworkName,
+					},
+					Spec: networkv1.NetworkSpec{
+						Type: "L3",
+						ParametersRef: &networkv1.NetworkParametersReference{
+							Group: Group,
+							Kind:  GKENetworkParamsKind,
+							Name:  DefaultGKENetworkParamsName,
+						},
+					},
+				},
+			},
+			gkeNwParams: []*networkv1.GKENetworkParams{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: DefaultGKENetworkParamsName,
+					},
+					Spec: networkv1.GKENetworkParamsSpec{
+						VPC:       DefaultVPCName,
+						VPCSubnet: DefaultVPCSubnetName,
+						PodIPv4Ranges: &networkv1.SecondaryRanges{
+							RangeNames: []string{DefaultSecondaryRangeA, DefaultSecondaryRangeB},
+						},
+					},
+				},
+			},
+			interfaces: []*compute.NetworkInterface{
+				{
+					AliasIpRanges: []*compute.AliasIpRange{
+						{IpCidrRange: "10.11.1.0/24", SubnetworkRangeName: DefaultSecondaryRangeA},
+					},
+					NetworkIP:  "80.1.172.1",
+					Network:    DefaultVPCName,
+					Subnetwork: DefaultVPCSubnetName,
+				},
+			},
+			wantDefaultNwPodCIDRs: []string{"10.11.1.0/24"},
+		},
+		{
+			desc: "one additional network along with default network",
```

Comment:
Would be good to add test cases where there are more networks present which don't match.

Reply:
Done.